libraries/core-react/src/components/Select/commonStyles.ts

```diff
@@ -0,0 +1,62 @@
+import { HTMLAttributes } from 'react'
+import { Input } from '../TextField/Input'
+import { List } from '../List'
+import { Button } from '../Button'
+import styled from 'styled-components'
+import { select as tokens } from './Select.tokens'
+import { typographyTemplate } from '@utils'
+
+const { ListItem } = List
+
+type ContainerProps = HTMLAttributes<HTMLDivElement>
+
+type StyledListItemType = {
+  highlighted: string
+}
+
+export const Container = styled.div<ContainerProps>`
+  position: relative;
+`
+
+export const PaddedInput = styled(Input)`
+  /* Hack: Had to add + 0px to satisfy the style lint plugin */
+  padding-right: calc(
+    ${tokens.button.size} + ${tokens.button.spacings.left} +
+      ${tokens.button.spacings.right} + 0px
+  );
+`
+
+export const StyledList = styled(List)`
+  background-color: ${tokens.background};
+  box-shadow: ${tokens.boxShadow};
+  overflow-y: scroll;
+  max-height: 300px;
+  padding: 0;
+  border-radius: ${tokens.borderRadius};
+  margin-top: 4px;
+  position: absolute;
+  right: 0;
+  left: 0;
+  z-index: 200;
```

Comment:
```suggestion
  z-index: 50;
```
I think Select option list should have lower priority than Popover (100) and Menu (150)